controllers/dns_helper.go

```diff
@@ -103,12 +104,12 @@ func gatewayDNSRecordLabels(gwKey client.ObjectKey) map[string]string {
 
 func (dh *dnsHelper) setEndpoints(mcgTarget *multicluster.GatewayTarget, dnsRecord *kuadrantdnsv1alpha1.DNSRecord, listener gatewayapiv1.Listener, strategy v1alpha1.RoutingStrategy) error {
 	gwListenerHost := string(*listener.Hostname)
-	var endpoints []*kuadrantdnsv1alpha1.Endpoint
+	var endpoints []*externaldns.Endpoint
 
 	//Health Checks currently modify endpoints so we have to keep existing ones in order to not lose health check ids
-	currentEndpoints := make(map[string]*kuadrantdnsv1alpha1.Endpoint, len(dnsRecord.Spec.Endpoints))
-	for _, endpoint := range dnsRecord.Spec.Endpoints {
-		currentEndpoints[endpoint.SetID()] = endpoint
+	currentEndpoints := make(map[string]*externaldns.Endpoint, len(dnsRecord.Spec.Endpoints))
+	for _, ep := range dnsRecord.Spec.Endpoints {
+		currentEndpoints[ep.SetIdentifier] = ep
 	}
```

Comment:
`SetIdentifier` could be empty which is why we were using the `SetID` method here. You would probably be better to copy that into here and use that instead to avoid any unexpected consequences https://github.com/Kuadrant/dns-operator/blob/v0.1.0/api/v1alpha1/dnsrecord_types.go#L65-L67

Reply:
ah ok, I'll do that now.